### PR TITLE
Code snippet analysis

### PR DESCRIPTION
**Pull request by zozs**
Implements support for code snippet analysis by calculating fingerprints.

Also upgrade dependencies to support Composer 2.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ docker build -t debricked/debricked-cli .
 
 ### Run tests
 All contributions are greatly welcome! To help you get started we have a included a
```
Comment (philomather) [inline README.md:73]:
```suggestion
All contributions are greatly welcome! To help you get started we have included a
```

File: src/Analysis/SnippetAnalysis.php [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+     * @param string $absoluteFilename the filename to analyse
+     * @param string $relativeFilename the filename to include in the WFP fingerprint
+     *
+     * @return bool true if the file was analysed, false if the file was ignored
+     */
+    public function analyseFile(string $absoluteFilename, string $relativeFilename): bool
+    {
+        // First determine if we want to scan it or not by checking the ignore list.
+        $extension = \pathinfo($absoluteFilename, PATHINFO_EXTENSION);
+        if (empty($extension) || \array_key_exists(\strtolower($extension), $this->ignore)) {
```
Comment (philomather) [inline src/Analysis/SnippetAnalysis.php:41]: Do we want this to analyse e.g. WORKSPACE files? They have no extension

File: src/Analysis/SnippetAnalysis.php [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+     * Returns the analysed snippets currently stored as a WFP file.
+     *
+     * @return string the WFP file as a string based on the current state
+     */
+    public function dumpWfp(): string
+    {
+        return implode($this->snippets);
+    }
+
+    // The following functions are from https://github.com/scanoss/scanner.php and is licensed under the MIT license,
```
Comment (philomather) [inline src/Analysis/SnippetAnalysis.php:65]:
```suggestion
    // The following functions are from https://github.com/scanoss/scanner.php and are licensed under the MIT license,
```

File: src/Command/FindAndUploadFilesCommand.php [hunk 4]
```diff
@@ -239,6 +250,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
                 $zip->addFile($absolutePathname, $relativePathname);
             }
 
+            // Include it in snippet analysis (SnippetAnalysis has it's own file extension filter though).
```
Comment (philomather) [inline src/Command/FindAndUploadFilesCommand.php:253]:
```suggestion
            // Include it in snippet analysis (SnippetAnalysis has its own file extension filter though).
```

File: src/Analysis/SnippetAnalysis.php (hunk 2 above)
Comment (zozs) [reply src/Analysis/SnippetAnalysis.php:41]: I would say that we don't care about them from a license perspective. I don't think they contain much code anyway. All files without extensions are also ignored in the upstream SCANOSS example integrations, so they don't seem to expect them to be scanned.

However, I'm open for discussions/suggestions regarding the list of ignored extensions, if you think anything needs to be changed.

File: src/Analysis/SnippetAnalysis.php [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    private array $ignore;
+
+    /** @var string[] */
+    private array $snippets;
+
+    public function __construct()
+    {
+        $this->ignore = ['bmp' => '', 'class' => '', 'conf' => '', 'csv' => '', 'eps' => '', 'gif' => '',
+            'gitignore' => '', 'gz' => '', 'jpeg' => '', 'jpg' => '', 'lock' => '', 'md' => '', 'mp3' => '',
+            'mp4' => '', 'pdf' => '', 'png' => '', 'rst' => '', 'sql' => '', 'tif' => '', 'zip' => '', ];
```
Comment (sweoggy) [inline src/Analysis/SnippetAnalysis.php:25]: `
```suggestion
        $this->ignore = ['ai' => '', 'apng' => '', 'avif' => '', 'bmp' => '', 'class' => '', 'conf' => '', 'csv' => '', 'dib' => '', 'eps' => '', 'gif' => '',
            'gitignore' => '', 'gz' => '', 'heic' => '', 'heif' => '', 'jpeg' => '', 'jpg' => '', 'jpe' => '', 'jif' => '', 'jfif' => '', 'jfi' => '', 'lock' => '', 'md' => '', 'mp3' => '',
            'mp4' => '', 'pdf' => '', 'png' => '', 'psd' => '', 'raw' => '', 'rst' => '', 'sql' => '', 'tif' => '', 'tiff' => '', 'webp' => '', 'zip' => '', ];
```

File: src/Analysis/SnippetAnalysis.php (hunk 2 above)
Comment (sweoggy) [reply src/Analysis/SnippetAnalysis.php:41]: Added a few image formats :)

**Pull request merged**